src/components/NewDesign/AnimatedButton/AnimatedButton.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import { Button } from '@material-ui/core'
+import successGif from '@static/gif/successAnimation.gif'
+import errorGif from '@static/gif/errorAnimation.gif'
+import useStyles from './style'
+
+export type ProgressState = 'progress' | 'approved' | 'success' | 'failed' | 'none'
+
+interface Props {
+  content: string
+  disabled?: boolean
+  progress: ProgressState
+  onClick: () => void
+}
+
+const AnimatedButton: React.FC<Props> = ({
+  content,
+  disabled = false,
+  progress,
+  onClick
+}) => {
+  const classes = useStyles()
+
+  const getMessage = () => {
+    if (progress === 'none') {
+      return <p className={classes.buttonContent}>{content}</p>
+    }
+
+    if (progress === 'progress' || progress === 'approved') {
+      return <p className={classes.buttonContent}>In progress..</p>
+    }
+
+    if (progress === 'success') {
+      return <img className={classes.gifContent} src={successGif}/>
+    }
+
+    if (progress === 'failed') {
```

Comment:
this last if is unnecessary, just return the last thing

Reply:
done